### PR TITLE
NO-TICKET-repair-post-bug

### PR DESCRIPTION
**Pull request by benvand**
* Accept 'no-reuse' instead of 0 as negative for reuse
* POST email_address instead of user object
* Update tests

See issue here for more details:
https://github.com/alphagov/digitalmarketplace-frontend-toolkit/issues/306

### CODE REVIEW

**Comment by TheDoubleK** [comment]
Looks good - nice catch that we don't need to pass the framework in the URL too.
:+1: once tests pass.

**Comment by benvand** [comment]
@TheDoubleK 
Also realised we were dealing with `supplier_framework` objects everywhere so made that clear and added a check to see that the declaration is complete before we allow reuse.
Seemed like the right thing to do.

File: app/main/helpers/frameworks.py [hunk 1]
```diff
@@ -104,13 +111,17 @@ def get_reusable_declaration(supplier_id, client, exclude_framework_slugs=None):
      :return: (declaration, framework)
      """
     exclude_framework_slugs = exclude_framework_slugs or []
-    declarations = {i['frameworkSlug']: i for i in client.find_supplier_declarations(supplier_id)['frameworkInterest']}
+    supplier_frameworks = client.find_supplier_declarations(supplier_id)['frameworkInterest']
+    supplier_frameworks_with_completed_declarations = filter(
```
Comment (benvand) [inline app/main/helpers/frameworks.py]: We are now :|

Comment (benvand) [inline app/main/helpers/frameworks.py]: Fortunately `get_supplier_framework_info` returns the declaration with the framework info so I swapped out `get_supplier_declaration` for `get_supplier_framework_info`

File: app/main/helpers/frameworks.py [hunk 2]
```diff
@@ -104,13 +104,17 @@ def get_reusable_declaration(supplier_id, client, exclude_framework_slugs=None):
      :return: (declaration, framework)
```
Comment (TheDoubleK) [inline app/main/helpers/frameworks.py]: Docstring needs to be updated as only returning the framework now.

**Comment by TheDoubleK** [comment]
Sorry, should have spotted this the first time round :/
Merge once Travis is happy :+1:

**Comment by TheDoubleK** [comment]
Going to merge this and fix the validation on the reuse page later.

**Pull request merged**